src/rust/engine/src/context.rs

```diff
@@ -35,12 +35,16 @@ impl Core {
     types: Types,
     build_root: PathBuf,
     ignore_patterns: Vec<String>,
+    work_dir: PathBuf,
   ) -> Core {
+    let mut snapshots_dir = work_dir.clone();
```

Comment:
err, tho I wonder if it makes sense to mutate `work_dir` in the event we want to use it for other work_dir subdir paths later within `Core`? from that perspective, I think the explicit clone makes sense - i.e. once `work_dir` == `.pants.d/snapshots` it no longer makes sense to refer to it as `work_dir` in my mind.